### PR TITLE
Merge_Sort.php added

### LINKED ISSUE #1632: Implementation of Merge Sort in PHP
The program for Merge Sort is not available in PHP. I would like to work on it! (GSSoC'20)
@singh-shreya6

### PR DESCRIPTION
**Pull request by goyalmuskan**
Fixes #1632 

#### Checklist:
- [x] 4 space indentation.
- [x] Coding conventions are followed.
- [ ] Input is taken dynamically.
- [x] Sample Input / Output is added at the end of file.
- [x] Logic Documentation (Comments).
- [x] File names are correct.

Make sure these boxes are checked before your pull request (PR) is ready to be reviewed and merged. Thanks!

#### Changes proposed in this pull request:
- added implementation of merge sort in php

#### Languages Used:
- PHP 

#### Files Added:
- Merge_Sort.php

> We're happy to help you get this ready -- don't be afraid to ask for help.

Thanks!

### CODE REVIEW

**Comment by goyalmuskan** [comment]
@singh-shreya6

**Comment by goyalmuskan** [comment]
@singh-shreya6

File: Merge_Sort/Merge_Sort.php [hunk 1]
```diff
@@ -0,0 +1,40 @@
+<?php
+/* Merge Sort is a Divide and Conquer algorithm. It divides input array in two halves, calls itself for the two halves and then merges the two sorted halves. */
+function merge_sort($my_array){
+	if(count($my_array) == 1 ) return $my_array;
+	$mid = count($my_array) / 2;
+    $left = array_slice($my_array, 0, $mid);
```
Comment (jainaman224) [inline Merge_Sort/Merge_Sort.php]: Inconsistent indentation.
Strictly follow 4 space indentation.

File: Merge_Sort/Merge_Sort.php [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+    $right = array_slice($my_array, $mid);
+	$left = merge_sort($left);
+	$right = merge_sort($right);
+	return merge($left, $right);
+}
+function merge($left, $right){
```
Comment (jainaman224) [inline Merge_Sort/Merge_Sort.php]: New line before a function.

**Comment by goyalmuskan** [comment]
Updated @jainaman224

File: Merge_Sort/Merge_Sort.php [hunk 3]
```diff
@@ -0,0 +1,43 @@
+<?php
+/* Merge Sort is a Divide and Conquer algorithm. It divides input array in two halves, calls itself for the two halves and then merges the two sorted halves. */
+function merge_sort($my_array){
+	if(count($my_array) == 1 ) return $my_array;
```
Comment (jainaman224) [inline Merge_Sort/Merge_Sort.php]: Use 4 space indentation instead of tab.

File: Merge_Sort/Merge_Sort.php [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+	$mid = count($my_array) / 2;
+        $left = array_slice($my_array, 0, $mid);
+        $right = array_slice($my_array, $mid);
+	$left = merge_sort($left);
+	$right = merge_sort($right);
+	return merge($left, $right);
+}
+
+function merge($left, $right){
```
Comment (jainaman224) [inline Merge_Sort/Merge_Sort.php]: Add space before `{`.

File: Merge_Sort/Merge_Sort.php [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+	}
+	while (count($right) > 0){
+		$res[] = $right[0];
+		$right = array_slice($right, 1);
+	}
+	return $res;
+}
+
+$test_array = array(100, 54, 7, 2, 5, 4, 1);
+echo implode(' ',merge_sort($test_array));
```
Comment (jainaman224) [inline Merge_Sort/Merge_Sort.php]: Add space after `,`.

**Comment by goyalmuskan** [comment]
@jainaman224 updated.

**Comment by goyalmuskan** [comment]
Please add labels to this PR @jainaman224 @asha15 :)

**Pull request merged**